diff --git a/node_modules/react-native/ReactAndroid/src/main/java/com/facebook/react/views/scroll/ReactScrollView.java b/node_modules/react-native/ReactAndroid/src/main/java/com/facebook/react/views/scroll/ReactScrollView.java
index 36c1576..efaa5f7 100644
--- a/node_modules/react-native/ReactAndroid/src/main/java/com/facebook/react/views/scroll/ReactScrollView.java
+++ b/node_modules/react-native/ReactAndroid/src/main/java/com/facebook/react/views/scroll/ReactScrollView.java
@@ -405,7 +405,8 @@ public class ReactScrollView extends ScrollView
     /* Offset from child's local coordinates to ScrollView coordinates */
     offsetDescendantRectToMyCoords(child, tempRect);
 
-    int scrollDelta = computeScrollDeltaToGetChildRectOnScreen(tempRect);
+    // int scrollDelta = computeScrollDeltaToGetChildRectOnScreen(tempRect);
+    int scrollDelta = computeScrollDeltaToGetChildRectAtTop(tempRect);
 
     if (scrollDelta != 0) {
       scrollBy(0, scrollDelta);
@@ -420,6 +421,20 @@ public class ReactScrollView extends ScrollView
     return super.computeScrollDeltaToGetChildRectOnScreen(rect);
   }
 
+  /**
+   * Experimental code to scroll a child view exactly to the top of the visible
+   * area of the scrollview
+   * @param rect
+   * @return Delta from current scroll position to the top of the child view
+   */
+  protected int computeScrollDeltaToGetChildRectAtTop(Rect rect) {
+    if (!mScrollEnabled) {
+      return 0;
+    }
+    int screenTop = getScrollY();
+    return rect.top - screenTop;
+  }
+
   @Override
   protected void onScrollChanged(int x, int y, int oldX, int oldY) {
     super.onScrollChanged(x, y, oldX, oldY);
